### PR TITLE
Add Validation Issues to Orchestrator

### PR DESCRIPTION
**Pull request by loic-sharma**
This updates the Orchestrator and its validators to use the new APIs that will be introduced in ServerCommon's v2.6 dependencies. In addition, the Orchestrator now persists `IValidationIssues` to the database.

**NOTE**: This will be updated with the real ServerCommon dependencies once https://github.com/NuGet/ServerCommon/pull/101 has been merged in.

### CODE REVIEW

**Comment by loic-sharma** [comment]
🔔 Ready for review

**Comment by loic-sharma** [comment]
Verified on DEV.

File: tests/NuGet.Services.Validation.Orchestrator.Tests/ValidationSetProcessorFacts.cs [hunk 1]
```diff
@@ -55,40 +56,40 @@ public class DefaultMockFacts : ValidationSetProcessorFactsBase
...
+                .ReturnsAsync(new ValidationResult(startStatus))
                 .Callback<IValidationRequest>(r => {
                     Assert.Equal(validation.Key, r.ValidationId);
                     actualStatus = startStatus;
                 })
                 .Verifiable();
 
             ValidationStorageMock
-                .Setup(vs => vs.MarkValidationStartedAsync(validation, startStatus))
+                .Setup(vs => vs.MarkValidationStartedAsync(validation, It.IsAny<ValidationResult>()))
```
Comment (agr) [inline tests/NuGet.Services.Validation.Orchestrator.Tests/ValidationSetProcessorFacts.cs]: if `startStatus` is dropped here, we need to add an assertion, that the status in result was actually the one we expect.

Comment (agr) [reply tests/NuGet.Services.Validation.Orchestrator.Tests/ValidationSetProcessorFacts.cs]: Or use `It.Is` with predicate that checks the status.

File: src/NuGet.Services.Validation.Orchestrator/ValidationStorageService.cs [hunk 2]
```diff
@@ -46,47 +46,71 @@ public ValidationStorageService(ValidationEntitiesContext validationContext, ILo
...
 
-            packageValidation.ValidationStatus = startedStatus;
+            packageValidation.ValidationStatus = validationResult.Status;
+
+            // If the validation has completed, save the validation issues to the package's validation.
+            if (validationResult.Status != ValidationStatus.Incomplete)
+            {
+                foreach (var validationIssue in validationResult.Issues)
+                {
+                    packageValidation.PackageValidationIssues.Add(new PackageValidationIssue
```
Comment (joelverhagen) [inline src/NuGet.Services.Validation.Orchestrator/ValidationStorageService.cs]: This code seems duplicated with below, right? can we have a helper method?

File: tests/NuGet.Services.Validation.Orchestrator.Tests/ValidationSetProcessorFacts.cs (hunk 1 above)
Comment (loic-sharma) [reply tests/NuGet.Services.Validation.Orchestrator.Tests/ValidationSetProcessorFacts.cs]: Good catch. I fixed this and a few other cases.

File: src/NuGet.Services.Validation.Orchestrator/ValidationStorageService.cs (hunk 2 above)
Comment (loic-sharma) [reply src/NuGet.Services.Validation.Orchestrator/ValidationStorageService.cs]: Fixed

**Comment by loic-sharma** [comment]
Depends on https://github.com/NuGet/ServerCommon/pull/105. @agr: you will need to update the dependencies here.

**Comment by agr** [comment]
Verified that it still works after the changes I made

File: src/NuGet.Services.Validation.Orchestrator/NuGet.Services.Validation.Orchestrator.csproj [hunk 3]
```diff
@@ -154,7 +154,7 @@
       <Version>4.3.0</Version>
     </PackageReference>
     <PackageReference Include="NuGetGallery.Core">
-      <Version>4.4.4-dev-18852</Version>
+      <Version>4.4.4-dev-new-msg-19655</Version>
```
Comment (joelverhagen) [inline src/NuGet.Services.Validation.Orchestrator/NuGet.Services.Validation.Orchestrator.csproj]: `dev-19677` should be usable now.

File: src/NuGet.Services.Validation.Orchestrator/ValidationOutcomeProcessor.cs [hunk 4]
```diff
@@ -56,7 +56,22 @@ public class ValidationOutcomeProcessor : IValidationOutcomeProcessor
...
                 {
                     await _galleryPackageService.UpdatePackageStatusAsync(package, PackageStatus.FailedValidation);
-                    _messageService.SendPackageValidationFailedMessage(package);
+
+                    var issuesExistAndAllPackageSigned = validationSet
+                        .PackageValidations
+                        .SelectMany(pv => pv.PackageValidationIssues)
+                        .Select(pvi => pvi.IssueCode == ValidationIssueCode.PackageIsSigned)
+                        .DefaultIfEmpty(false)
+                        .All(v => v);
```
Comment (joelverhagen) [inline src/NuGet.Services.Validation.Orchestrator/ValidationOutcomeProcessor.cs]: I wish there was an `.All()` for `bool`...

**Comment by loic-sharma** [comment]
Tested on DEV

**Pull request merged**